helix-term/src/ui/mod.rs

```diff
@@ -95,7 +95,15 @@ pub fn regex_prompt(
 
                             view.ensure_cursor_in_view(doc, config.scrolloff);
                         }
-                        Err(_err) => (), // TODO: mark command line as error
+                        Err(_err) => {
+                            if event == PromptEvent::Validate {
+                                let error = format!("Invalid regex: {}", input);
+                                cx.editor.set_error(error);
```

Comment:
If it errors out we should also restore the initial cursor position, identical to what happens in the `PrompEvent::Abort` block a few lines up.